native/core/src/execution/operators/filter.rs

```diff
@@ -62,6 +65,8 @@ pub struct FilterExec {
     default_selectivity: u8,
     /// Properties equivalence properties, partitioning, etc.
     cache: PlanProperties,
+    /// The projection indices of the columns in the output schema of join
+    projection: Option<Vec<usize>>,
```

Comment:
is this projection a part of migration? if so the migration is quite complicated....

Reply:
We currently maintain a copy of DataFusion's `FilterExec` with one small change, so I copied over that latest to keep in sync and then re-applied the change that we need (for memory safety because of the way we re-use buffers).